libevmasm/SimplificationRule.h

```diff
@@ -55,84 +56,106 @@ struct SimplificationRule
-	static auto constexpr ADDRESS = Instruction::ADDRESS;
-	static auto constexpr BALANCE = Instruction::BALANCE;
-	static auto constexpr ORIGIN = Instruction::ORIGIN;
-	static auto constexpr CALLER = Instruction::CALLER;
-	static auto constexpr CALLVALUE = Instruction::CALLVALUE;
-	static auto constexpr CALLDATALOAD = Instruction::CALLDATALOAD;
-	static auto constexpr CALLDATASIZE = Instruction::CALLDATASIZE;
-	static auto constexpr CALLDATACOPY = Instruction::CALLDATACOPY;
-	static auto constexpr CODESIZE = Instruction::CODESIZE;
-	static auto constexpr CODECOPY = Instruction::CODECOPY;
-	static auto constexpr GASPRICE = Instruction::GASPRICE;
-	static auto constexpr EXTCODESIZE = Instruction::EXTCODESIZE;
-	static auto constexpr EXTCODECOPY = Instruction::EXTCODECOPY;
-	static auto constexpr RETURNDATASIZE = Instruction::RETURNDATASIZE;
-	static auto constexpr RETURNDATACOPY = Instruction::RETURNDATACOPY;
-	static auto constexpr EXTCODEHASH = Instruction::EXTCODEHASH;
-	static auto constexpr BLOCKHASH = Instruction::BLOCKHASH;
-	static auto constexpr COINBASE = Instruction::COINBASE;
-	static auto constexpr TIMESTAMP = Instruction::TIMESTAMP;
-	static auto constexpr NUMBER = Instruction::NUMBER;
-	static auto constexpr DIFFICULTY = Instruction::DIFFICULTY;
-	static auto constexpr GASLIMIT = Instruction::GASLIMIT;
-	static auto constexpr CHAINID = Instruction::CHAINID;
-	static auto constexpr SELFBALANCE = Instruction::SELFBALANCE;
-	static auto constexpr POP = Instruction::POP;
-	static auto constexpr MLOAD = Instruction::MLOAD;
-	static auto constexpr MSTORE = Instruction::MSTORE;
-	static auto constexpr MSTORE8 = Instruction::MSTORE8;
-	static auto constexpr SLOAD = Instruction::SLOAD;
-	static auto constexpr SSTORE = Instruction::SSTORE;
-	static auto constexpr PC = Instruction::PC;
-	static auto constexpr MSIZE = Instruction::MSIZE;
-	static auto constexpr GAS = Instruction::GAS;
-	static auto constexpr LOG0 = Instruction::LOG0;
-	static auto constexpr LOG1 = Instruction::LOG1;
-	static auto constexpr LOG2 = Instruction::LOG2;
-	static auto constexpr LOG3 = Instruction::LOG3;
-	static auto constexpr LOG4 = Instruction::LOG4;
-	static auto constexpr CREATE = Instruction::CREATE;
-	static auto constexpr CALL = Instruction::CALL;
-	static auto constexpr CALLCODE = Instruction::CALLCODE;
-	static auto constexpr STATICCALL = Instruction::STATICCALL;
-	static auto constexpr RETURN = Instruction::RETURN;
-	static auto constexpr DELEGATECALL = Instruction::DELEGATECALL;
-	static auto constexpr CREATE2 = Instruction::CREATE2;
-	static auto constexpr REVERT = Instruction::REVERT;
-	static auto constexpr INVALID = Instruction::INVALID;
-	static auto constexpr SELFDESTRUCT = Instruction::SELFDESTRUCT;
+
+	template<Instruction inst>
+	struct PatternGenerator
+	{
+		template<typename... Args> constexpr Pattern operator()(Args&&... _args) const
+		{
+			return {inst, std::initializer_list<Pattern>{std::forward<Args>(_args)...}};
```

Comment:
```suggestion
			return {inst, {std::forward<Args>(_args)...}};
```
I think that could be removed now - also below (not that it matters too much).